### ISSUE TITLE
Certain errors thrown from within `waitUntil` cannot be caught

### ISSUE THREAD

**Issue by diachedelic** [webdriverio/webdriverio#3815]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.7.8
 - **Mode:** WDIO testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 8.12.0
 - **NPM version:** 6.9.0
 - **Browser name and version:** 73.0.3683.86 (Official Build) (64-bit)
 - **Platform name and version:** MacOS 10.14.3
 - **Additional wdio packages used (if applicable):** 

```
    "@wdio/allure-reporter": "^5.7.8",
    "@wdio/cli": "^5.7.8",
    "@wdio/local-runner": "^5.7.8",
    "@wdio/mocha-framework": "^5.7.8",
    "@wdio/selenium-standalone-service": "^5.7.8",
    "@wdio/spec-reporter": "^5.7.8",
    "@wdio/sync": "^5.7.8",
```

**Config of WebdriverIO**

```javascript
exports.config = {
    runner: 'local',
    specs: ['./test/**/*.js'],
    exclude: [],
    maxInstances: 10,
    capabilities: [
        {
            maxInstances: 5,
            browserName: 'chrome',            
            'goog:chromeOptions': {
                args: [
                    '--headless', 
                    '--disable-gpu'
                ]                
            }
        },
    ],
    logLevel: 'error',
    deprecationWarnings: true,
    bail: 0,
    baseUrl: baseUrl,
    waitforTimeout: 10000,
    connectionRetryTimeout: 90000,
    connectionRetryCount: 3,
    services: ['selenium-standalone'],
    framework: 'mocha',
    reporters: ['spec','allure'],
    mochaOpts: {
        ui: 'bdd',
        timeout: 60000,
        compilers: []
    },
}
```

**Describe the bug**

Certain errors thrown from within `waitUntil` cannot be caught with an outer `try...catch` block.

```javascript
try {
  browser.waitUntil(() => {
    $('#not_exist').getAttribute('class')
  },)
} catch (err) {
  // swallow error
}
```

```
[chrome #0-0] Error: Can't call getAttribute on element with selector "#not_exist" because element wasn't found
[chrome #0-0]     at <anonymous>
[chrome #0-0]     at Browser.browser.waitUntil (/Users/me/Code/WebdriverIO-TypeScript-Boilerplate/test/example.spec.js:15:25)
```

At first I thought it was related to #3479 , but maybe not.

**To Reproduce**
```
git clone git@github.com:diachedelic/WebdriverIO-TypeScript-Boilerplate.git &&
  cd WebdriverIO-TypeScript-Boilerplate &&
  npm install &&
  npm test # I think java downloads itself so be patient
```

**Expected behavior**

_Any_ errors thrown from within `waitUntil` should be catchable by a `try...catch` block containing `waitUntil`.

**Log**
[wdio.log](https://gist.github.com/diachedelic/9a0cb2bdd57e18e978ed4c758f4faf51)

**Comment by diachedelic** [comment]
My workaround, if anyone is interested, it to swallow the error inside `waitUntil`, and let `waitUntil` throw a timeout error.

```javascript
    browser.waitUntil(() => {
      // wait for transition class to be added
      try {
        $('#not_exist').getAttribute('class')
      } catch (err) {
        return false
      }
    }, 5000)
```

**Comment by diachedelic** [comment]
Turns out this also affects custom commands, using `addCommand`, see repro for example:

```javascript
    browser.addCommand('throwElementNotFoundError', () => {
      $('#not_exist').getAttribute('class')
    }, true)

    try {
      browser.$('body').throwElementNotFoundError()
    } catch (err) {
      // swallow error
    }
```

Error is not swallowed.

**Comment by diachedelic** [comment]
Timeout errors also escape custom commands:

```javascript
    browser.addCommand('throwTimeoutError', () => {
      browser.waitUntil(() => false, 1000)
    })

    try {
      browser.throwTimeoutError()
    } catch (err) {
      // should swallow error
    }
```

**Comment by christian-bromann** [comment]
Fixed in #3816

**Comment by diachedelic** [comment]
I can confirm this is fixed in v5.7.9, thanks for the quick response!
